File: app/src/api/delete-post.js
import { useWorkspace } from '@/composables'

export const DeletePost = async (thread) => {
    const { wallet, program } = useWorkspace()
    await program.value.rpc.DeletePost({
        accounts: {
            author: wallet.value.publicKey,
            thread: thread.publicKey,
        },
    })
}

File: programs/grape-eve/src/instructions/thread/delete_post.rs
use crate::states::thread::Thread;
use anchor_lang::prelude::*;

#[derive(Accounts)]
pub struct DeletePost<'info> {
    #[account(mut, has_one = author, close = author)]
    pub thread: Account<'info, Thread>,
    pub author: Signer<'info>,
}

pub fn delete_post(_ctx: Context<DeletePost>) -> Result<()> {
    Ok(())
}
